api/agent.go

```diff
@@ -60,10 +60,10 @@ func (a *Agent) populateCache(self *AgentSelf) {
 		a.nodeName = self.Member.Name
 	}
 	if a.datacenter == "" {
-		a.datacenter, _ = self.Member.Tags["dc"]
+		a.datacenter, _ = self.Config["Datacenter"].(string)
```

Comment:
Technically this code is panic-safe, but it's overly clever so I'll fix it :)

(The 2 form assignment applies to the type assertion, so if the key doesn't exist it returns `interface{}(nil)` which when 2-form-assignment-asserted to string gives you the empty string which datacenter is anyway! Way too "clever" :) https://play.golang.org/p/5jcXxfdthi )